File: src/extern/src/components/aircraft/Aircraft.cpp
#include "Aircraft.h"

#include "models/Aircraft.h"
#include "services/servicelocator/ServiceLocator.h"

namespace components {

std::vector<models::Aircraft> Aircraft::LoadAircraftInCircle(const models::GeoPoint &center, int radius) {
  return std::vector<models::Aircraft>();
}

std::vector<models::Aircraft> Aircraft::LoadCurrentOrderAircraft() {
  return std::vector<models::Aircraft>();
}

std::vector<models::AircraftClass> Aircraft::LoadAircraftClasses() {
  return services::ServiceLocator::GetDataManager().LoadAircraftClasses();
}

std::vector<models::Pad> Aircraft::LoadNearestPads(const models::GeoPoint &position) {
  return std::vector<models::Pad>();
}

}

File: src/handlers/aircraft.py
import sys
import json
import web
from extern import aircraft
from helpers import comma_separated_to_float_tuple


class BaseAircraftHandler(web.BaseHandlerWithExternModule):
    def get_extern_module(self):
        return aircraft


class LoadInCircleHandler(BaseAircraftHandler):
    def post(self):
        try:
            data = json.loads(self.request.body)
            center = tuple(data.get("center"))
            radius = int(data.get("radius"))
            self.write(self.extern_wrapper.load_aircraft_in_circle(center, radius))
        except Exception:
            sys.stderr.write(str(Exception))


class LoadCurrentOrderAircraftHandler(BaseAircraftHandler):
    def get(self):
        self.write(self.extern_wrapper.load_current_order_aircraft())


class LoadAircraftClassesHandler(BaseAircraftHandler):
    def get(self):
        self.write(self.extern_wrapper.load_aircraft_classes())


class LoadNearestPadsHandler(BaseAircraftHandler):
    def post(self):
        try:
            data = json.loads(self.request.body)
            position = tuple(data.get("position"))
            self.write(self.extern_wrapper.load_nearest_pads(position))
        except Exception:
            sys.stderr.write(str(Exception))
